logstash-core/src/main/java/org/logstash/settings/DeprecatedAlias.java

```diff
@@ -33,19 +35,36 @@ public final class DeprecatedAlias<T> extends SettingDelegator<T> {
 
     private static final DeprecationLogger DEPRECATION_LOGGER = new DefaultDeprecationLogger(LOGGER);
 
-    private SettingWithDeprecatedAlias<T> canonicalProxy;
+    private final SettingWithDeprecatedAlias<T> canonicalProxy;
+
+    private final String obsoletedVersion;
 
-    DeprecatedAlias(SettingWithDeprecatedAlias<T> canonicalProxy, String aliasName) {
+    DeprecatedAlias(SettingWithDeprecatedAlias<T> canonicalProxy, String aliasName, String obsoletedVersion) {
         super(canonicalProxy.getCanonicalSetting().deprecate(aliasName));
         this.canonicalProxy = canonicalProxy;
+        this.obsoletedVersion = obsoletedVersion;
     }
 
     // Because loggers are configure after the Settings declaration, this method is intended for lazy-logging
     // check https://github.com/elastic/logstash/pull/16339
     public void observePostProcess() {
         if (isSet()) {
-            DEPRECATION_LOGGER.deprecated("The setting `{}` is a deprecated alias for `{}` and will be removed in a " +
-                    "future release of Logstash. Please use `{}` instead", getName(), canonicalProxy.getName(), canonicalProxy.getName());
+            String dmsg = "The setting `{}` is a deprecated alias for `{}`";
+            ArrayList<Object> params = new ArrayList<>();
+            params.add(getName());
+            params.add(canonicalProxy.getName());
+
+            if (this.obsoletedVersion != null && !this.obsoletedVersion.isEmpty()) {
+                dmsg += " and will be removed in version {}.";
+                params.add(this.obsoletedVersion);
+            } else {
+                dmsg += " and will be removed in a future release of Logstash.";
+            }
+
+            dmsg += " Please use `{}` instead";
+            params.add(canonicalProxy.getName());
+
+            DEPRECATION_LOGGER.deprecated(dmsg, params.toArray());
```

Comment:
Since we're already appending text to `dmsg`, is there any point to still use parameters here?

Reply:
changed to string builder without params. thank you.